src/Compilers/CSharp/Portable/Binder/Binder_Initializers.cs

```diff
@@ -152,42 +152,34 @@ internal struct ProcessedFieldInitializers
                     var syntaxRef = initializer.Syntax;
                     Debug.Assert(syntaxRef.SyntaxTree.Options.Kind != SourceCodeKind.Regular);
 
-                    var initializerNode = (CSharpSyntaxNode)syntaxRef.GetSyntax();
-
-                    if (binderFactory == null)
-                    {
-                        binderFactory = compilation.GetBinderFactory(syntaxRef.SyntaxTree);
-                    }
-
-                    Binder scriptClassBinder = binderFactory.GetBinder(initializerNode);
-                    Debug.Assert(((ImplicitNamedTypeSymbol)scriptClassBinder.ContainingMemberOrLambda).IsScriptClass);
-
-                    if (firstDebugImports == null)
+                    if (parentBinder == null)
                     {
-                        firstDebugImports = scriptClassBinder.ImportChain;
+                        var syntaxTree = syntaxRef.SyntaxTree;
+                        var binderFactory = compilation.GetBinderFactory(syntaxTree);
+                        var compilationUnit = (CompilationUnitSyntax)syntaxTree.GetRoot();
```

Comment:
Can we ever get here during speculative binding?  If so, can we still assume the root is a ```CompilationUnitSyntax```?